### DISCUSSION TITLE
Terraform modules without patch version like v1.0 are skipped

### DISCUSSION THREAD

**Discussion by rbojan** [renovatebot/renovate#20551] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

renovate/renovate:34.147.0-slim@sha256:998921cb81e8060ef7407b1164763ec1b2c2ff208e5782178eccb5dfd90be450

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### If you're self-hosting Renovate, tell us what version of the platform you run.

15.8.2

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Describe the bug

Our Terraform modules are versioned like `v0.1` or `v1.0` and hosted by GitLab.

We reference the modules like this:

```tf
module "renovate_test" {
  source = "git::https://github.com/rbojan/terraform-renovate-test.git?ref=v1.0"
}
```

Renovate stopped updating the dependencies saying `"currentValue: "v1.0"` `"skipReason": "invalid-value"` (more in the attached logs).

### Steps to reproduce

1. Create main.tf where the module is used:

```tf
module "renovate_test" {
  source = "git::https://github.com/rbojan/terraform-renovate-test.git?ref=v1.0"
}
```

2. Run renovate

3. Expected change: v1.0 -> v1.1
  Current behaviour: skipped

### Relevant debug logs

<details><summary>Logs</summary>

```
               {
                 "currentValue": "v1.0",
                 "datasource": "git-tags",
                 "depIndex": 2,
                 "depName": "github.com/rbojan/terraform-renovate-test",
                 "depType": "module",
                 "packageName": "https://github.com/rbojan/terraform-renovate-test.git",
                 "skipReason": "invalid-value",
                 "versioning": "semver",
                 "warnings": [],
                 "updates": []
               },
```

</details>

### Have you created a minimal reproduction repository?

I have linked to a minimal reproduction repository in the bug description

**Comment by HonkingGoose** [comment]
Hi @rbojan 👋 

I can't find the link to your minimal reproduction in the issue. Can you say where we can find the reproduction?

**Comment by viceice** [comment]
probably wrong versioning assigned , or no versioning at all, so fallback to semver which fails because those versions are not compatible.

**Comment by rbojan** [comment]
> Hi @rbojan 👋
> 
> I can't find the link to your minimal reproduction in the issue. Can you say where we can find the reproduction?

Sorry, I thought that a module repo  https://github.com/rbojan/terraform-renovate-test and **Steps to reproduce** are enough.

Currently we only have a renovate-runner instance running in a private environment. What is the fastest way to run renovate in a public environment for minimal reproduction?

**Comment by HonkingGoose** [comment]
> What is the fastest way to run renovate in a public environment for minimal reproduction?

It's easiest for us if you can reproduce the problem in a public GitHub repository. We can then fork the repository and run Renovate on the fork.

Read the [minimal reproductions guide](https://github.com/renovatebot/renovate/blob/main/docs/development/minimal-reproductions.md) for the full instructions.

**Comment by rbojan** [comment]
Here we go: https://github.com/rbojan/terraform-renovate-test-version

**Comment by secustor** [comment]
> probably wrong versioning assigned , or no versioning at all, so fallback to semver which fails because those versions are not compatible.

It is a fall back to `semver`, as there is no versioning defined in `git-tags`.

**Comment by secustor** [comment]
@rbojan  Overwriting the versioning to `semver-coerced` should work for you as it isn't enforcing semver2 as strictly.  

If you want to overwrite this for all of your Terraform modules, you can add this snippet to your packageRules: 

```json
{
  "packageRules": [
    {
      "matchManagers": ["terraform"],
      "matchDepTypes": ["module"],
      "versioning": "semver-coerced"
    }
  ]
}
```

**Comment by viceice** [comment]
converted to discussion as it's not a bug. it's currently by design.

**Comment by rbojan** [comment]
@secustor It works, but not as expected:

Expected change: `v1.0` -> `v1.1`
Current behaviour: `v1.0` -> `v1.1.3`

Renovate pins to the latest PATCH version instead of MINOR. 

Is this by design of the `semver-coerced` versioning?

**Reply by rbojan** [reply]
According to https://github.com/renovatebot/renovate/blob/main/lib/modules/versioning/semver-coerced/readme.md this is by design.

Is there a solution for what we are trying to do?

Thank you for your time and help!

**Reply by secustor** [reply]
So I guess, you have a flowing tag `v1.1` which is currently on the same commit as `v1.1.3`?

**Reply by rbojan** [reply]
Correct.

**Reply by rarkins** [reply]
Docker versioning?

**Reply by secustor** [reply]
@rbojan  As @rarkins  has suggested, you could try to use `docker` versioning in that case. We have there some extra logic as this is a common case for docker images. 

```json
{
  "packageRules": [
    {
      "matchManagers": ["terraform"],
      "matchDepTypes": ["module"],
      "versioning": "docker"
    }
  ]
}
```

**Reply by rbojan** [reply]
Works, thank you!

**Comment by secustor** [comment, marked as answer]
```json
{
  "packageRules": [
    {
      "matchManagers": ["terraform"],
      "matchDepTypes": ["module"],
      "versioning": "docker"
    }
  ]
}
```
Reposting answer for better discoverability. Original comment: https://github.com/renovatebot/renovate/discussions/20551#discussioncomment-5075917
